src/main/java/seedu/address/model/person/Module.java

```diff
@@ -4,15 +4,15 @@
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
 /**
- * Represents a Person's address in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ * Represents a Person's module in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidModule(String)}
  */
-public class Address {
+public class Module {
 
     public static final String MESSAGE_CONSTRAINTS = "Addresses can take any values, and it should not be blank";
 
     /*
-     * The first character of the address must not be a whitespace,
+     * The first character of the module must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
     public static final String VALIDATION_REGEX = "[^\\s].*";
```

Comment:
try this regex, "^[A-Z]{2}\d{4}" then test case try with CS2100, CSS200, C21000, only 1st one should be true

Reply:
I left the regex as such because we won't know how other module codes are like. For example, I've taken courses that are labelled GET1031, with 3 letter prefixes. We also have modules such as CS2030S, with a one-letter prefix at the end. I will try to think of a better regex to follow module codes more strictly, thanks for the mention!

Reply:
Oh right, I forgot about those cases, maybe try "[A-Z]{2-3}[0-9]{4}[A-Z]{1}", should take into account of the 3 letters at the front and a letter at the back